packages/core/admin/admin/src/utils/urls.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * @description Creates an absolute URL, if there is no URL or it
+ * is relative, we use the `window.location.origin` as a fallback.
+ * IF it's an absolute URL, we return it as is.
+ */
+const createAbsoluteUrl = (url?: string): string => {
+  if (!url) {
+    return window.location.origin;
+  }
+  if (url.startsWith('/')) {
```

Comment:
Just a note: this will catch protocol-relative URLS (like `//strapi.com/foo`) and then ignore window.location.origin since it already has a base URL.

So ultimately it will arrive at the correct place -- if someone puts backend url as "//myserver" `new URL toString` afaik will return "//myserver" despite hitting this line -- but probably not intentionally, so it might be worth commenting that, or adding an explicit check for that case

Reply:
I've added a test case and a comment 🤓